File: JavaLibrary/src/test/java/com/flat/simplex/parser/logic/block/BlockDefaultTest.java
package com.flat.simplex.parser.logic.block;

import com.flat.simplex.parser.logic.Context;
import com.flat.simplex.parser.logic.error.Error;
import com.flat.simplex.support.TokenChain;
import org.junit.jupiter.api.Test;

import static com.flat.simplex.support.ContextSupport.assertErrors;
import static com.flat.simplex.support.TokenChain.mChain;
import static com.flat.simplex.support.TokenChain.readChain;

class BlockDefaultTest {

    @Test
    public void blockDefault() {
        TokenChain chain = readChain("default:");

        Context context = new Context(chain.get());
        BlockSwitch blockSwitch = getBlockSwitch(context);
        BlockDefault blockDefault = new BlockDefault(blockSwitch, chain.get(), null);
        blockDefault.read();

        assertErrors(context);
    }

    @Test
    public void blockDefaultWithoutSwitch_Fail() {
        TokenChain chain = readChain("default:");

        Context context = new Context(chain.get());
        BlockDefault blockDefault = new BlockDefault(context, chain.get(), null);
        blockDefault.read();
        assertErrors(context, Error.defaultOutOfPlace);
    }

    @Test
    public void blockDefaultUnexpectedToken_Fail() {
        TokenChain chain = readChain("default hello:");

        Context context = new Context(chain.get());
        BlockSwitch blockSwitch = getBlockSwitch(context);
        BlockDefault blockDefault = new BlockDefault(blockSwitch, chain.get(), null);
        blockDefault.read();

        assertErrors(context, Error.unexpectedToken);
    }

    @Test
    public void blockDefaultUnexpectedEndOfTokens_Fail() {
        TokenChain chain = readChain("default");

        Context context = new Context(chain.get());
        BlockSwitch blockSwitch = getBlockSwitch(context);
        BlockDefault blockDefault = new BlockDefault(blockSwitch, chain.get(), null);
        blockDefault.read();

        assertErrors(context, Error.unexpectedEndOfTokens);
    }

    private BlockSwitch getBlockSwitch(Context context) {
        TokenChain chain = readChain("switch(value){}");

        return new BlockSwitch(context, chain.get(), null);
    }
}

File: SimplexApp/simplex/parser/logic/block/BlockDefault.test.js
const TokenChain = require("simplex/support/TokenChain.js");
const Context = require("simplex/parser/logic/Context.js");
const Error = require("simplex/parser/logic/error/Error.js");
const BlockDefault = require("simplex/parser/logic/block/BlockDefault.js");
const BlockSwitch = require("simplex/parser/logic/block/BlockSwitch.js");
const ContextSupport = require("simplex/support/ContextSupport.js");

const readChain = TokenChain.readChain;
const assertErrors = ContextSupport.assertErrors;

test('blockDefault', () => {
    let chain = readChain('default:');
    let context = new Context(chain.get());
    let blockSwitch = getBlockSwitch(context);
    let blockDefault = new BlockDefault(blockSwitch, chain.get(), null);
    blockDefault.read();
    assertErrors(context);
});
test('blockDefaultWithoutSwitch_Fail', () => {
    let chain = readChain('default:');
    let context = new Context(chain.get());
    let blockDefault = new BlockDefault(context, chain.get(), null);
    blockDefault.read();
    assertErrors(context, Error.defaultOutOfPlace);
});
test('blockDefaultUnexpectedToken_Fail', () => {
    let chain = readChain('default hello:');
    let context = new Context(chain.get());
    let blockSwitch = getBlockSwitch(context);
    let blockDefault = new BlockDefault(blockSwitch, chain.get(), null);
    blockDefault.read();
    assertErrors(context, Error.unexpectedToken);
});
test('blockDefaultUnexpectedEndOfTokens_Fail', () => {
    let chain = readChain('default');
    let context = new Context(chain.get());
    let blockSwitch = getBlockSwitch(context);
    let blockDefault = new BlockDefault(blockSwitch, chain.get(), null);
    blockDefault.read();
    assertErrors(context, Error.unexpectedEndOfTokens);
});
function getBlockSwitch(context) {
    let chain = readChain('switch(value){}');
    return new BlockSwitch(context, chain.get(), null);
}